ingest/ledgerbackend/captive_core_backend.go

```
@@ -146,21 +146,35 @@ func NewCaptive(config CaptiveCoreConfig) (*CaptiveStellarCore, error) {
 	var cancel context.CancelFunc
 	config.Context, cancel = context.WithCancel(parentCtx)
 
-	archive, err := historyarchive.Connect(
-		config.HistoryArchiveURLs[0],
-		historyarchive.ConnectOptions{
-			NetworkPassphrase:   config.NetworkPassphrase,
-			CheckpointFrequency: config.CheckpointFrequency,
-			Context:             config.Context,
-		},
-	)
+	var validArchives []historyarchive.ArchiveInterface
+
+	for _, url := range config.HistoryArchiveURLs {
+		archive, err := historyarchive.Connect(
+			url,
+			historyarchive.ConnectOptions{
+				NetworkPassphrase:   config.NetworkPassphrase,
+				CheckpointFrequency: config.CheckpointFrequency,
+				Context:             config.Context,
+			},
+		)
+
+		if err != nil {
+			config.Log.Warnf("Error connecting to history archive (%s): %s", url, err)
+			continue
+		}
+
+		validArchives = append(validArchives, archive)
+	}
+
+	pool, err := historyarchive.CreatePool(validArchives...)
 	if err != nil {
 		cancel()
-		return nil, errors.Wrap(err, "error connecting to history archive")
+		config.Log.Error("Error connecting to ALL history archives.")
```

I think you can return a wrapped error `errors.Wrap(err, "Error connecting to ALL history archives.")` instead of logging the error